app/controllers/payments_controller.rb

```diff
@@ -1,22 +1,49 @@
 class PaymentsController < ApplicationController
-
   before_action :ensure_current_order_present
-  before_action :ensure_checkout_allowed
-  before_action :update_current_order_state
+  before_action :ensure_checkout_allowed, only: :new
+  before_action :update_current_order_state, only: :new
+  before_action :build_payment, only: :create
+  before_action :create_stripe_payment, only: :create
+  before_action :complete_order, only: :create
 
   def new
   end
 
+  def create
+    redirect_to order_details_path(invoice: @payment.order.invoice_number)
+  end
+
+
   private
 
     def update_current_order_state
       if current_order.cart?
-        redirect_to cart_path, alert: I18n.t(:address_not_added, scope: [:flash, :alert])
+        redirect_to cart_path, alert: I18n.t(:address_not_added, scope: [:flash, :alert]) and return
 
       elsif current_order.can_pay?
         unless current_order.pay
-          redirect_to cart_path, alert: current_order.pretty_base_errors
+          redirect_to cart_path, alert: current_order.pretty_base_errors and return
         end
       end
     end
+
+    def build_payment
+      @payment = current_order.build_payment
```

Comment:
use `payments.build` directly and move it `create` action.
No need for a before action here